chapters/part3/fairness.md

```diff
@@ -722,9 +722,289 @@ We have made the [bias-mitigated, finetuned large RoBERTA model for SNLI](https:
+
+`AdversarialBiasMitigator` works with any Transformer predictor that has a static embedding layer. The relationships between (1) the number of epochs of finetuning, (2) the number of layers in the predictor, (3) the type of bias direction used, and the efficacy of bias mitigation is a promising direction for future research. While finetuning for more epochs with `AdversarialBiasMitigator` might push the predictor to depend less on gender, it could also cause the predictor to overfit to spurious correlations and/or discover proxies for gender that are not contained within the gender bias direction. Furthermore, it is likely harder to mitigate biases for predictors with a larger number of layers, as they are capable of learning extremely complex mappings and achieving convergence in large adversarial networks is tricky (this is further discussed in <b>Pitfalls of Adversarial Bias Mitigation</b> below). Finally, how well `AdversarialBiasMitigator` works strongly depends on the finetuning data and predictor architecture. You should not assume that using a `AdversarialBiasMitigator` resolves any particular bias issue in a predictor. There is published literature suggesting that it should help (without noticeably compromising predictor performance on the test set), but that needs to be evaluated for any actual downstream use.
+
+2. While the adversary's parameters are updated normally during finetuning, the predictor's parameters are updated such that the predictor will not aid the adversary and will make it more difficult for the adversary to recover binary gender bias. Formally, we update the predictor's parameters according to Equation 1 from [Mitigating Unwanted Biases with Adversarial Learning](https://api.semanticscholar.org/CorpusID:9424845). The projection term in Equation 1 ensures that the predictor doesn't accidentally aid the adversary and the final term in Equation 1 serves to update the predictor's parameters to increase the adversary's loss.
+
+![adversarial_equation_1](/part3/fairness/adversarial_equation_1.png)
+
+We achieve this non-traditional parameter update scheme in AllenNLP using an [`on_backward`](http://docs.allennlp.org/main/api/training/callbacks/callback/#on_backward) trainer callback. An [`on_backward`](http://docs.allennlp.org/main/api/training/callbacks/backward/) trainer callback performs custom backpropagation and allows for gradient manipulation; `on_backward` callback(s) will be executed in lieu of the default `loss.backward()` called by `GradientDescentTrainer`.
+
+We provide the [`AdversarialBiasMitigatorBackwardCallback`](http://docs.allennlp.org/main/api/fairness/adversarial_bias_mitigator/#adversarialbiasmitigatorbackwardcallback) to perform the non-traditional parameter updates required by adversarial bias mitigation.
+
+```python
+@TrainerCallback.register("adversarial_bias_mitigator_backward")
```

Comment:
Perhaps it makes sense to just have the snippet with the actual parameter update in the guide; we already provide a link to the complete code.